pkg/comp-functions/functions/vshnmariadb/mariadb_deploy.go

```diff
@@ -0,0 +1,263 @@
+	cpu := comp.Spec.Parameters.Size.CPU
+	disk := comp.Spec.Parameters.Size.Disk
+
+	if reqMem == "" {
+		reqMem = resouces.MemoryRequests.String()
+	}
+	if reqCPU == "" {
+		reqCPU = resouces.CPURequests.String()
+	}
+	if mem == "" {
+		mem = resouces.MemoryLimits.String()
+	}
+	if cpu == "" {
+		cpu = resouces.CPULimits.String()
+	}
+	if disk == "" {
+		disk = resouces.Disk.String()
+	}
+
+	values := map[string]interface{}{
+		"fullnameOverride": comp.GetName(),
+		"replicaCount":     1,
+		"resources": map[string]interface{}{
+			"requests": map[string]interface{}{
+				"memory": reqMem,
+				"cpu":    reqCPU,
+			},
+			"limits": map[string]interface{}{
+				"memory": mem,
+				"cpu":    cpu,
+			},
+		},
+		"tls": map[string]interface{}{
+			"enabled":            true,
+			"certificatesSecret": "tls-server-certificate",
+			"certFilename":       "tls.crt",
+			"certKeyFilename":    "tls.key",
+			"certCAFilename":     "ca.crt",
+		},
+		"mariadbConfiguration": comp.Spec.Parameters.Service.MariadbSettings,
+		"persistence": map[string]interface{}{
+			"size":         disk,
+			"storageClass": comp.Spec.Parameters.StorageClass,
+		},
+		"startupProbe": map[string]interface{}{
+			"enabled": true,
+		},
+		"metrics": map[string]interface{}{
+			"enabled": true,
+			"containerSecurityContext": map[string]interface{}{
+				"enabled": false,
+			},
+		},
+		"securityContext": map[string]interface{}{
+			"enabled": false,
+		},
+		"containerSecurityContext": map[string]interface{}{
+			"enabled": false,
+		},
+		"podSecurityContext": map[string]interface{}{
+			"enabled": false,
+		},
+	}
+
+	vb, err := json.Marshal(values)
+	if err != nil {
+		err = fmt.Errorf("cannot marshal helm values: %w", err)
+		return err
+	}
+
+	r := &xhelmbeta1.Release{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: comp.GetName(),
+		},
+		Spec: xhelmbeta1.ReleaseSpec{
+			ForProvider: xhelmbeta1.ReleaseParameters{
+				Chart: xhelmbeta1.ChartSpec{
+					Repository: svc.Config.Data["chartRepository"],
+					Version:    svc.Config.Data["chartVersion"],
+					Name:       "mariadb-galera",
+				},
+				Namespace: comp.GetInstanceNamespace(),
+				ValuesSpec: xhelmbeta1.ValuesSpec{
+					Values: k8sruntime.RawExtension{
+						Raw: vb,
+					},
+				},
+			},
+			ResourceSpec: xpv1.ResourceSpec{
+				ProviderConfigReference: &xpv1.Reference{
+					Name: "helm",
+				},
+				WriteConnectionSecretToReference: &xpv1.SecretReference{
+					Name:      comp.GetName() + "-connection",
+					Namespace: comp.GetInstanceNamespace(),
+				},
+			},
+			ConnectionDetails: []xhelmbeta1.ConnectionDetail{
+				{
+					ObjectReference: corev1.ObjectReference{
+						APIVersion: "v1",
+						Kind:       "Secret",
+						Name:       comp.GetName(),
+						Namespace:  comp.GetInstanceNamespace(),
+						FieldPath:  "data.mariadb-root-password",
+					},
+					ToConnectionSecretKey: "MARIADB_PASSWORD",
+				},
+				{
+					ObjectReference: corev1.ObjectReference{
```

Comment:
It doesn't look like the user can actually change the port, so we could just use a const string`mysqlPort` with the value `"3306"` for now and expose that directly in the connection details.  `svc.SetConnectionDetail("MARIADB_PORT",[]byte(mysqlPort)`

Reply:
Used const instead